#!/usr/bin/env python
PACKAGE = "phidgets"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("braking_strength", double_t, 0, "The braking strength to apply when the duty cycle is 0", 4, 0, 6)
gen.add("data_interval_ms", int_t, 0, "The number of milliseconds between acquisitions of data on the device", 4, 0, 6)
gen.add("publish_rate", int_t, 0, "How often the driver will publish data on the ROS topic. If 0 (the default), it will publish every time there is an update from the device (so at the data_interval_ms). If positive, it will publish the data at that rate regardless of the acquisition interval", 4, 0, 6)

exit(gen.generate(PACKAGE, "phidgets", "Motor"))